File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/security/CheckFileMessage.scala





















// Generated on 05/03/2013 10:35:01
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait CheckFileMessage extends MessageDeserializer {
  val messageId = 6156

  protected trait Message extends BaseMessage with Serializable {
    val messageId = CheckFileMessage.messageId

    val filenameHash: String
    val theType: Byte
    val value: String

    def serialize(buf: Buffer) {
      buf.writeString(filenameHash)
      buf.writeByte(theType)
      buf.writeString(value)
    }

  }

  protected def create(filenameHash: String, theType: Byte, value: String): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val filenameHash = buf.readString()
    val theType = buf.readByte()
    val value = buf.readString()

    create(filenameHash, theType, value)
  }

}

object CheckFileMessage extends CheckFileMessage {
  case class MessageImpl(filenameHash: String, theType: Byte, value: String) extends Message
  type Target = MessageImpl

  override def create(filenameHash: String, theType: Byte, value: String) = MessageImpl(filenameHash, theType, value)

  def apply(filenameHash: String, theType: Byte, value: String) = create(filenameHash, theType, value)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/CheckFileMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class CheckFileMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6156;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String filenameHash;
    public byte type;
    public String value;
    
    public CheckFileMessage() { }
    
    public CheckFileMessage(String filenameHash, byte type, String value) {
        this.filenameHash = filenameHash;
        this.type = type;
        this.value = value;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(filenameHash);
        buf.writeByte(type);
        buf.writeString(value);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        filenameHash = buf.readString();
        type = buf.readByte();
        if (type < 0)
            throw new RuntimeException("Forbidden value on type = " + type + ", it doesn't respect the following condition : type < 0");
        value = buf.readString();
    }
    
}
